src/System.Console/src/System/IO/SyncTextReader.cs

```diff
@@ -8,10 +8,13 @@
 
 namespace System.IO
 {
+    /* SyncTextReader intentionally locks on itself rather than a private lock object.
+     * This is done to ensure colored console readers can be synchronized with non-colored
+     * readers (Issue#2855).
+     */
```

Comment:
Umm...you are right. In fact even for the writer's it can potentially impact other states (for example, encoding, although it is not exposed right now). I will remove the color from the comments.